posts/2021/06/pytorch-ignite-distributed.rst

```diff
@@ -0,0 +1,460 @@
+      </div>
+   </embed>
+
+.. note::
+  Additionally, ``idist`` provides collective operations like
+  ``all_reduce``, ``all_gather``, and ``broadcast`` that can be used
+  with all supported distributed frameworks. Please, see `our
+  documentation <https://pytorch.org/ignite/distributed.html#ignite-distributed-utils>`__
+  for more details.
+
+Examples
+========
+
+The code snippets below highlight the API's specificities of each of the
+distributed backends on the same use case as compared to the ``idist``
+API. Torch native code is available for DDP, Horovod, and for XLA/TPU
+devices.
+
+PyTorch-Ignite's unified code snippet can be run with the standard Torch
+backends like ``gloo`` and ``nccl`` and also with Horovod and XLA for
+TPU devices. Note that the code is less verbose, however, the user still
+has full control of the training loop.
+
+The following examples are introductory. For a more robust,
+production-grade example that uses PyTorch-Ignite, refer
+`here <https://github.com/pytorch/ignite/tree/master/examples/contrib/cifar10>`__.
+
+The complete source code of these experiments can be found
+`here <https://github.com/pytorch-ignite/idist-snippets>`__.
+
+PyTorch-Ignite - Torch native Distributed Data Parallel - Horovod - XLA/TPUs
+----------------------------------------------------------------------------
+
+.. raw:: html
+
+   <embed> 
+      <div>
+         <table>
+            <tr>
+               <th style="text-align:center; padding: 0;">
+                  <h3><b><u>PyTorch-Ignite</u></b></h3></th>
+               <th style="text-align:center; padding: 0;">
+                  <h3><b><u>PyTorch DDP</u></b></h3></th>
+            </tr>
+            <tr>
+               <td style="text-align:center; padding: 0;"> <a href="https://github.com/pytorch-ignite/idist-snippets/blob/master/ignite_idist.py"><h3>Source Code</h3></a> </th>
+               <td style="text-align:center; padding: 0;"> <a href="https://github.com/pytorch-ignite/idist-snippets/blob/master/torch_native.py"><h3>Source Code</h3></a> </th>
+            </tr>
+            <tr>
+               <td colspan="2"> <img src="/images/pytorch-ignite/distributed-made-easy-with-ignite/ignite_vs_ddp_whole.png"> </td>
+            </tr>
+            <tr>
+               <th style="text-align:center;">
+                  <h3><b><u>Horovod</u></b></h3></th>
+               <th style="text-align:center;">
+                  <h3><b><u>Torch XLA</u></b></h3></th>
+            </tr>
+            <tr>
+               <td style="text-align:center;"> <a href="https://github.com/pytorch-ignite/idist-snippets/blob/master/torch_horovod.py"><h3>Source Code</h3></a> </th>
+               <td style="text-align:center;"> <a href="https://github.com/pytorch-ignite/idist-snippets/blob/master/torch_xla_native.py"><h3>Source Code</h3></a> </th>
+            </tr>
+            <tr>
+               <td> <img src="/images/pytorch-ignite/distributed-made-easy-with-ignite/horovod_whole.png"> </td>
```

Comment:
The `horovod_whole.png` image has the wrong contents, it actually contains code for torch native DDP (i.e. image 2 and image 3 here are the same).